### PR TITLE
[ci] Improve analysis_options alignment with flutter/packages

### PR DESCRIPTION
**Pull request by stuartmorgan**
Updates analysis options to reduce diff from flutter/packages (which also means reducing diff from flutter/flutter since that repo had a more recent re-alignment). The only code changes needed (other than a handful of local ignores of new options) was to remove a lot of unnecessary awaits in returns.

This also includes a small fix to the way the repo tooling's `update-release-info` works to avoid unnecessary CHANGELOG updates when run against this PR (https://github.com/flutter/flutter/issues/113941).

Part of https://github.com/flutter/flutter/issues/76229 and https://github.com/flutter/flutter/issues/113764
Fixes https://github.com/flutter/flutter/issues/113941

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
This pulled in a ton of people due to CODEOWNERS, but it just needs one review from whoever gets to it since the actual plugin changes are trivial and mechanical.

File: packages/camera/camera_web/example/integration_test/camera_web_test.dart
Comment (hellohuanlin) [inline packages/camera/camera_web/example/integration_test/camera_web_test.dart:1667]: is await optional here? are these functions still async?

**Review by hellohuanlin — APPROVED** [review]
left a question

File: packages/camera/camera_web/example/integration_test/camera_web_test.dart
Comment (stuartmorgan) [reply packages/camera/camera_web/example/integration_test/camera_web_test.dart:1667]: Everything is still async; see https://dart-lang.github.io/linter/lints/unnecessary_await_in_return.html for a full example of the lint that these changes are fixing.

Comment (hellohuanlin) [reply packages/camera/camera_web/example/integration_test/camera_web_test.dart:1667]: fascinating!

**Pull request merged**
